### PR TITLE
Rename client identifier from Satoshi to MagicBean (closes #1481)

### LINKED ISSUE #1481: Change the client identifier
from "Satoshi" to something we haven't decided on yet. Suggestions welcome.

### PR DESCRIPTION
**Pull request by bitcartel**
@ageis seeder will have be to updated too.

### CODE REVIEW

File: src/rpcnet.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ Value getpeerinfo(const Array& params, bool fHelp)
             "    \"pingtime\": n,             (numeric) ping time\n"
             "    \"pingwait\": n,             (numeric) ping wait\n"
             "    \"version\": v,              (numeric) The peer version, such as 7001\n"
```
Comment (daira) [inline src/rpcnet.cpp]: Change to ``The peer's protocol version, such as 170002``.

File: src/rpcnet.cpp [hunk 2]
```diff
@@ -97,7 +97,7 @@ Value getpeerinfo(const Array& params, bool fHelp)
...
-            "    \"subver\": \"/Satoshi:0.8.5/\",  (string) The string version\n"
+            "    \"subver\": \"/MagicBean:0.8.5/\",  (string) The string version\n"
```
Comment (daira) [inline src/rpcnet.cpp]: Change to ``/MagicBean:1.0.0/``

File: src/rpcnet.cpp [hunk 3]
```diff
@@ -387,7 +387,7 @@ Value getnetworkinfo(const Array& params, bool fHelp)
             "\nResult:\n"
             "{\n"
             "  \"version\": xxxxx,                      (numeric) the server version\n"
-            "  \"subversion\": \"/Satoshi:x.x.x/\",     (string) the server subversion string\n"
+            "  \"subversion\": \"/MagicBean:x.x.x/\",     (string) the server subversion string\n"
```
Comment (daira) [inline src/rpcnet.cpp]: Change to ``/MagicBean:x.y.z[-v]/``

File: contrib/seeds/makeseeds.py [hunk 4]
```diff
@@ -27,7 +27,7 @@
 PATTERN_IPV4 = re.compile(r"^((\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})):(\d+)$")
 PATTERN_IPV6 = re.compile(r"^\[([0-9a-z:]+)\]:(\d+)$")
 PATTERN_ONION = re.compile(r"^([abcdefghijklmnopqrstuvwxyz234567]{16}\.onion):(\d+)$")
-PATTERN_AGENT = re.compile(r"^(\/Satoshi:0\.8\.6\/|\/Satoshi:0\.9\.(2|3|4|5)\/|\/Satoshi:0\.10\.\d{1,2}\/|\/Satoshi:0\.11\.\d{1,2}\/)$")
+PATTERN_AGENT = re.compile(r"^(\/MagicBean:0\.8\.6\/|\/MagicBean:0\.9\.(2|3|4|5)\/|\/MagicBean:0\.10\.\d{1,2}\/|\/MagicBean:0\.11\.\d{1,2}\/)$")
```
Comment (daira) [inline contrib/seeds/makeseeds.py]: Should be something like:
```
PATTERN_AGENT = re.compile(r"^(\/MagicBean:[-.0-9a-z]+\/)$")
```

**Comment by daira** [comment]
NACK, this comment blocks: https://github.com/zcash/zcash/pull/1514/files#r83026389

**Comment by bitcartel** [comment]
@daira Updated commit with push  -f.  Review comments have been addressed.

File: contrib/seeds/makeseeds.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 PATTERN_IPV4 = re.compile(r"^((\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})):(\d+)$")
 PATTERN_IPV6 = re.compile(r"^\[([0-9a-z:]+)\]:(\d+)$")
 PATTERN_ONION = re.compile(r"^([abcdefghijklmnopqrstuvwxyz234567]{16}\.onion):(\d+)$")
-PATTERN_AGENT = re.compile(r"^(\/Satoshi:0\.8\.6\/|\/Satoshi:0\.9\.(2|3|4|5)\/|\/Satoshi:0\.10\.\d{1,2}\/|\/Satoshi:0\.11\.\d{1,2}\/)$")
+PATTERN_AGENT = re.compile(r"^(\/MagicBean:([0-9a-z-.]+)\/)$")
```
Comment (daira) [inline contrib/seeds/makeseeds.py]: I'm pretty sure it doesn't matter, but why have an inner capturing group when there wasn't one before?

**Comment by daira** [comment]
utACK

File: contrib/seeds/makeseeds.py (hunk 5 above)
Comment (bitcartel) [inline contrib/seeds/makeseeds.py]: We'll probably never need it but just in case we need to extract the version identifier.

**Comment by bitcartel** [comment]
Passes tests.  ACK. @zkbot r+

**Pull request merged**